File: count_factors.cpp
#include <iostream>
#include <map>
using namespace std;

int count(int n)
{
    map<int, int> factors;
    int count = 1;

    int i = 2;
    while (i * i <= n)
    {
        while (n % i == 0)
        {
            factors[i]++;
            n /= i;
        }
        i++;
    }

    if (n > 1)
    {
        factors[n]++;
    }

    for (auto &pair : factors)
    {
        count *= (pair.second + 1);
    }

    return count;
}

int main()
{
    int n;
    cin >> n;
    for (int i = 0; i < n; ++i)
    {
        int num;
        cin >> num;
        cout << count(num) << endl;
    }

    return 0;
}

File: count_factors.py
from collections import defaultdict

def count(n):
    factors = defaultdict(int)
    count = 1

    i = 2
    while i * i <= n:
        while n%i == 0:
            factors[i] += 1
            n //= i
        
        i += 1

    if n > 1:
        factors[i] += 1

    for val in factors.values():
        count *= (val + 1)
    
    return count

n = int(input())
for _ in range(n):
    num = int(input())
    print(count(num))
    